finetune/input_pipeline.py

```diff
@@ -12,18 +13,37 @@
 from sklearn.model_selection import train_test_split
 
 
+import finetune
 from finetune.errors import FinetuneError
 from finetune.config import PAD_TOKEN
-from finetune.encoding import TextEncoder, ArrayEncodedOutput, EncodedOutput
+from finetune.base_models.gpt.encoder import GPTEncoder
+from finetune.base_models.gpt2.encoder import GPT2Encoder
+from finetune.encoding import ArrayEncodedOutput, EncodedOutput
 from finetune.imbalance import compute_class_weights
 
-ENCODER = TextEncoder()
 LOGGER = logging.getLogger('finetune')
 
+ENCODERS = {
```

Comment:
Not sure this is the responsibility of input pipeline, can we move this inside base_models so we don't have to import them here.

Reply:
:+1: